flexget/plugins/filter/movie_queue.py

```diff
@@ -396,3 +432,275 @@ def queue_get(session=None, downloaded=False):
+        'imdb_id': {'type': 'string', 'pattern': r'tt\d{7}'},
+        'tmdb_id': {'type': 'integer'},
+        'movie_id': {'type': 'integer'}
+    },
+    'anyOf': [
+        {'required': ['title']},
+        {'required': ['imdb_id']},
+        {'required': ['tmdb_id']},
+        {'required': ['movie_id']}
+    ]
+}
+
+movie_del_results_schema = api.schema('movie_del_results_schema', movie_del_results_schema)
+movie_del_input_schema = api.schema('movie_del_input_schema', movie_del_input_schema)
+
+movie_edit_results_schema = {
+    'type': 'object',
+    'properties': {
+        'message': {'type': 'string'},
+        'movie': movie_object
+    }
+}
+
+movie_edit_input_schema = {
+    'type': 'object',
+    'properties': {
+        'quality': {'type': 'string', 'format': 'quality_requirements'},
+        'reset_downloaded': {'type': 'boolean', 'default': False}
+    },
+    'anyOf': [
+        {'required': ['quality']},
+        {'required': ['reset_downloaded']}
+    ]
+}
+
+movie_edit_results_schema = api.schema('movie_edit_results_schema', movie_edit_results_schema)
+movie_edit_input_schema = api.schema('movie_edit_input_schema', movie_edit_input_schema)
+
+
+@movie_queue_api.route('/')
+class MovieQueueAPI(APIResource):
+    @api.response(404, 'Page does not exist')
+    @api.response(200, 'Movie queue retrieved successfully', movie_queue_schema)
+    @api.doc(parser=movie_queue_parser)
+    def get(self, session=None):
+        """ List queued movies """
+        args = movie_queue_parser.parse_args()
+        page = args['page']
+        max_results = args['max']
+        downloaded = args['downloaded']
+        sort_by = args['sort_by']
+        order = args['order']
+
+        movie_queue = queue_get(session=session, downloaded=downloaded)
+        count = len(movie_queue)
+
+        pages = int(ceil(count / float(max_results)))
+
+        movie_items = []
+
+        if page > pages and pages != 0:
+            return {'error': 'page %s does not exist' % page}, 404
+
+        start = (page - 1) * max_results
+        finish = start + max_results
+        if finish > count:
+            finish = count
+
+        for movie_number in range(start, finish):
+            movie_items.append(movie_queue[movie_number].to_dict())
+
+        sorted_movie_list = sorted(movie_items, key=itemgetter(sort_by), reverse=order)
+
+        return jsonify({
+            'movies': sorted_movie_list,
+            'number_of_movies': count,
+            'page_number': page,
+            'total_number_of_pages': pages
+        })
+
+    @api.response(400, 'Page not found')
+    @api.response(200, 'Movie successfully added', movie_add_results_schema)
+    @api.validate(movie_add_input_schema)
+    def post(self, session=None):
+        """ Add movies to movie queue """
+        kwargs = request.json
+
+        try:
+            kwargs['quality'] = qualities.Requirements(kwargs.get('quality'))
+        except ValueError as e:
```

Comment:
This should never happen if you set quality as required in movie_add_input_schema.. so there would be no need for this try except